src/control.cc

```diff
@@ -93,6 +93,20 @@ void ControlModule::for_eval(AbstractNode &node, const ModuleInstantiation &inst
 				for_eval(node, inst, l+1, &c, evalctx);
 			}
 		}
+                else if (it_values->type() == Value::ValueType::STRING) {
+			const std::string val = it_values->toString();
+			const char *ptr = val.c_str();
+
+			gchar outbuf[8];
+			const glong len = g_utf8_strlen(ptr, -1);
+			for (;*ptr && len > 0;ptr = g_utf8_next_char(ptr)) {
+				const gunichar ch = g_utf8_get_char(ptr);
+				const gint end = g_unichar_to_utf8(ch, outbuf);
+				outbuf[end] = 0;
```

Comment:
I wonder how much overhead there is to this. This may just be premature optimization, but would smth. like this be more efficient? ..similarly for other occurrences of such conversion from utf-8 to unicode and back to utf-8
```
  do {
    auto next = g_utf8_next_char(ptr);
    c.set_variable(it_name, ValuePtr(std::string{ptr, next-ptr}));
    for_eval(node, inst, l+1, &c, evalctx);
    ptr = next;
  } while (ptr);
```

Reply:
I don't think that works, `std::string{ptr, next-ptr}` would copy the encoded representation in utf-8, not the code point value?

E.g. [Unicode Character 'GRINNING FACE' (U+1F600)](http://www.fileformat.info/info/unicode/char/1f600/index.htm) would generate `0xF0 0x9F 0x98 0x80 / 0xf09f9880) instead of `0x0001F600`.

Reply:
..but in the code above, don't we start with UTF8, convert one character to unicode, then convert back to UTF8? My suggestion was to skip the conversion and just copy the UTF-8 bytes directly.

Reply:
Ahh, right, the char value is not used here as with `ord()` but just converted back to string. Let me try that...

Reply:
Looks great. In terms of safety, I'm not sure what will happen for invalid strings. calling `g_utf8_validate()` prior to running such loops may be recommended.

Reply:
Yep, although it's probably better to push the guaranties into the parser so we don't have to validate on every evaluation.